### PR TITLE
add net.casimirlab.frigoligo

### PR DESCRIPTION
**Pull request by casimir**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
> Frigoligo is a [wallabag](https://wallabag.org/) client crafted using [Flutter](https://flutter.dev/). Its core philosophy is to prioritize pragmatism and adhering to best practices when it makes sense.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/casimir/frigoligo/graphs/contributors
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: net.casimirlab.frigoligo.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - tar -xvf frigoligo-v1.0.10-linux-x64.tar.gz -C /app/
+      - chmod +x /app/frigoligo/frigoligo
+      - ln -s /app/frigoligo/frigoligo /app/bin/frigoligo
+      - mkdir -p /app/share/icons/hicolor/512x512/apps /app/share/applications /app/share/metainfo
+      - cp play_store_512.png /app/share/icons/hicolor/512x512/apps/net.casimirlab.frigoligo.png
+      - cp net.casimirlab.frigoligo.desktop /app/share/applications/
+      - cp net.casimirlab.frigoligo.metainfo.xml /app/share/metainfo/
+    sources:
+      - type: file
+        url: https://github.com/casimir/frigoligo/releases/download/v1.0.10/frigoligo-v1.0.10-linux-x64.tar.gz
```
Comment (hfiguiere) [inline net.casimirlab.frigoligo.yml:32]: so how will that work on aarch64?

File: net.casimirlab.frigoligo.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      - x86_64
+    build-commands:
+      - mkdir -p /app/bin
+      - tar -xvf frigoligo-v1.0.10-linux-x64.tar.gz -C /app/
+      - chmod +x /app/frigoligo/frigoligo
+      - ln -s /app/frigoligo/frigoligo /app/bin/frigoligo
+      - mkdir -p /app/share/icons/hicolor/512x512/apps /app/share/applications /app/share/metainfo
+      - cp play_store_512.png /app/share/icons/hicolor/512x512/apps/net.casimirlab.frigoligo.png
+      - cp net.casimirlab.frigoligo.desktop /app/share/applications/
+      - cp net.casimirlab.frigoligo.metainfo.xml /app/share/metainfo/
```
Comment (hfiguiere) [inline net.casimirlab.frigoligo.yml:29]:
```suggestion
      - install -Dm644 play_store_512.png /app/share/icons/hicolor/512x512/apps/net.casimirlab.frigoligo.png
      - install -Dm644 net.casimirlab.frigoligo.desktop -t /app/share/applications/
      - install -Dm644 net.casimirlab.frigoligo.metainfo.xml -t /app/share/metainfo/
```

File: net.casimirlab.frigoligo.desktop
Comment (hfiguiere) [inline net.casimirlab.frigoligo.desktop:1]: this should be upstream.

File: net.casimirlab.frigoligo.metainfo.xml
Comment (hfiguiere) [inline net.casimirlab.frigoligo.metainfo.xml:1]: this should be upstream

File: net.casimirlab.frigoligo.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>net.casimirlab.frigoligo</id>
```
Comment (hfiguiere) [inline net.casimirlab.frigoligo.metainfo.xml:3]: `casimirlab.net` doesn't see to exist as a domain

yet:

> [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines](https://docs.flathub.org/docs/for-app-authors/requirements#application-id) are followed.

File: net.casimirlab.frigoligo.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  <description>
+    <p>
+      Take your links anywhere and keep them fresh! Frigoligo is a wallabag client striving to be simple but also fast and easy to use. Just like wallabag, Frigoligo is free to use, ad-free and open source.
+    </p>
+    <p>
+      You need a wallabag instance to use this application.
+    </p>
+  </description>
+  
+  <content_rating type="oars-1.0" />
```
Comment (hfiguiere) [inline net.casimirlab.frigoligo.metainfo.xml:32]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: net.casimirlab.frigoligo.yml (hunk 1 above)
Comment (casimir) [reply net.casimirlab.frigoligo.yml:32]: It doesn't. I though the following lines prevent this case. https://github.com/flathub/flathub/blob/e75f33ea3014d273dc641138177ef07a3460d24a/net.casimirlab.frigoligo.yml#L19-L20

Comment (casimir) [reply net.casimirlab.frigoligo.yml:31]: I misunderstood the documentation then and that was not the intention. I can't find anything about architecture support, is it mandatory to support both `x86_64` and `aarch64`?

Comment (casimir) [reply net.casimirlab.frigoligo.yml]: This has been changed.

File: net.casimirlab.frigoligo.metainfo.xml
Comment (casimir) [reply net.casimirlab.frigoligo.metainfo.xml:1]: done

**Comment by casimir** [comment]
I think everything has been addressed.

**Comment by casimir** [comment]
bot build net.casimirlab.frigoligo

**Comment by casimir** [comment]
bot, build net.casimirlab.frigoligo

File: net.casimirlab.frigoligo.metainfo.xml (hunk 3 above)
Comment (bbhtt) [reply net.casimirlab.frigoligo.metainfo.xml]: I can't reach it still.

Also do you have control over the whole domain or only the subdomain?

File: net.casimirlab.frigoligo.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/flatpak/flatpak-builder/main/data/flatpak-manifest.schema.json
+
+app-id: net.casimirlab.frigoligo
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
```
Comment (bbhtt) [inline net.casimirlab.frigoligo.yml:5]: 24.08 is latest

File: net.casimirlab.frigoligo.metainfo.xml (hunk 3 above)
Comment (bbhtt) [reply net.casimirlab.frigoligo.metainfo.xml]: https://casimirlab.net/ needs to be reachable HTTP 200 you aren't using the subdomain in the app ID.

Comment (casimir) [reply net.casimirlab.frigoligo.metainfo.xml]: I have the control of the whole domain but I only use subdomains for my services.

I have set up a static server that return a 204 for https://casimirlab.net and a well known file for the subsequent verification at https://casimirlab.net/.well-known/org.flathub.VerifiedApps.txt

Does that work for you?

File: net.casimirlab.frigoligo.yml (hunk 5 above)
Comment (casimir) [reply net.casimirlab.frigoligo.yml]: Updated in the lastest commit.

**Comment by casimir** [comment]
bot, build net.casimirlab.frigoligo

File: net.casimirlab.frigoligo.metainfo.xml (hunk 3 above)
Comment (bbhtt) [reply net.casimirlab.frigoligo.metainfo.xml]: yes

**Comment by casimir** [comment]
Awesome! Thanks @hfiguiere @bbhtt for your reviews.

**Pull request closed without merging**